src/MainDashBoard/components/drawer/DrawerModal.tsx

```diff
@@ -0,0 +1,272 @@
+import Modal from '@components/Modal';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import useModal from '@hooks/useModal';
+import { ComponentProps, FocusEvent, FormEvent, useId, useState } from 'react';
+
+interface IModalInputProps extends ComponentProps<'input'> {
+  isActive: boolean;
+  label: string;
+}
+
+/** 날짜 입력 받는 인풋창 */
+const ModalInput = ({ isActive, label, ...props }: IModalInputProps) => {
+  const uniqueId = useId();
+  const isYear = props.placeholder?.length === 4;
+  const InputComponent = isYear ? YearInput : DateInput;
+
+  return (
+    <>
+      <InputComponent
+        required
+        id={uniqueId}
+        $activeExtend={isActive}
+        disabled={!isActive}
+        name={props.name}
+        value={props.value}
+        defaultValue={props.defaultValue}
+        type="text"
+        placeholder={props.placeholder}
+        pattern={props.pattern}
+        maxLength={props.maxLength}
+        title={props.title}
+        onChange={props.onChange}
+        onBlur={props.onBlur}
+      />
```

Comment:
props로 받는건 구조분해 할당하면 어떤가요?

```suggestion
const {name, value, defaultValue, placeholder, pattern, maxLength, title, onChange, onBlur } = props

  return (
    <>
      <InputComponent
        required
        id={uniqueId}
        $activeExtend={isActive}
        disabled={!isActive}
        name={name}
        value={value}
        defaultValue={defaultValue}
        type="text"
        placeholder={placeholder}
        pattern={pattern}
        maxLength={maxLength}
        title={title}
        onChange={onChange}
        onBlur={onBlur}
      />
```

Reply:
완료했습니다 :)